src/app/profiles/profile-detail/profile-detail.component.ts

```diff
@@ -305,11 +306,47 @@ export class ProfileDetailComponent implements OnInit {
 
   confirm (): void {
     // Warn user of risk if using random generated passwords
+    // Warn user of risk is CIRA configuration and static network are selected simultaneously
     if (this.profileForm.valid) {
       this.isLoading = true
       const result: any = Object.assign({}, this.profileForm.getRawValue())
       // When creating new
-      if (!this.isEdit && (result.generateRandomPassword || result.generateRandomMEBxPassword)) {
+      if ((result.connectionMode === 'CIRA' && result.dhcpEnabled === false) &&
```

Comment:
Might be a good idea to create functions for each dialog that is shown to help break up the code a bit.